pkgs/servers/nosql/mongodb/mongodb.nix

```diff
@@ -11,17 +30,49 @@ with lib;
 }:
 
 let
-  variants = if versionAtLeast version "4.2"
-    then rec { python = scons.python.withPackages (ps: with ps; [ pyyaml cheetah3 psutil setuptools ]);
-            scons = sconsPackages.scons_3_1_2.override { python = python3; }; # 4.2 < mongodb <= 5.0.x needs scons 3.x built with python3
-            mozjsVersion = "60";
-            mozjsReplace = "defined(HAVE___SINCOS)";
-          }
-    else rec { python = scons.python.withPackages (ps: with ps; [ pyyaml typing cheetah ]);
-            scons = sconsPackages.scons_3_1_2;
-            mozjsVersion = "45";
-            mozjsReplace = "defined(HAVE_SINCOS)";
-          };
+  variants =
+    if versionAtLeast version "6.0" then rec {
+      python = scons.python.withPackages (ps: with ps; [
+        pyyaml
+        cheetah3
+        psutil
+        setuptools
+        packaging
+        pymongo
+      ]);
+
+      # 4.2 < mongodb <= 6.0.x needs scons 3.x built with python3
+      scons = sconsPackages.scons_3_1_2.override { python = python3; };
+
+      mozjsVersion = "60";
+      mozjsReplace = "defined(HAVE___SINCOS)";
+
+    } else if versionAtLeast version "4.2" then rec {
+      python = scons.python.withPackages (ps: with ps; [
+        pyyaml
+        cheetah3
+        psutil
+        setuptools
+      ]);
+
+      # 4.2 < mongodb <= 5.0.x needs scons 3.x built with python3
+      scons = sconsPackages.scons_3_1_2.override { python = python3; };
+
+      mozjsVersion = "60";
+      mozjsReplace = "defined(HAVE___SINCOS)";
```

Comment:
Do we still need this option? It is the same everywhere.

Reply:
It's different for versions <4.2 (one underscore instead of three for
`mozjsReplace` for example).

Dropping the old versions should simplify things but this is out of the scope
of this pull request.

Reply:
If we add a new mongodb version we might as well drop one really old one to keep the variants low.

Reply:
#172009 seems to favour marking the old versions as deprecated instead to still
offer some upgrade path:
https://github.com/NixOS/nixpkgs/pull/172009/files#diff-127f942ecb068c0807b0f2b48f7ccc949d1236061766447c4af79250a6da2662